Refurbished Withings Body Plus Notebook

cthev456 opened the pull request:
The main goal of this task was to refurbish the original Withings Body Plus notebook to utilize the Wearipedia package and append additional sections to match the standard set by the Garmin Fenix notebook.

cthev456 committed:
Add files via upload

cthev456 committed:
Refurbished original notebook

alrojo commented:
**General comments**

The notebook is high quality for sure, but please refrain from directly copying anything from the Garmin notebook (paragraphs, plots). Instead, you should read it as guidance. We'd like all plots to be different from the other notebooks.

There's a few warnings throughout the code, please resolve them.

**Introduction**

There's no readme, instead link to http://wearipedia.com/wearables/withings-body-plus
"In this guide, we sequentially cover the following five topics to extract from the Withings API:" it's 9 topics

Does step 1-9 truly reflect the guide, or is it copied from the Garmin? Ensure that the sub bulletpoints reflect the writings in the guide

**1.Setup**

1.1 this should be written in a letter format

1.2 The artificial data should not be generated in the beginning from a gist, it should be generated using the Wearipedia package as done in the garmin notebook

wearipedia python package shouldn't be imported until the data extraction part, see the Garmin notebook

**2 Auth**

It has steps 1, 2, 3, 4, and 7. Where's step 5, 6?

**3 Data extraction**

"The overall health data API page" hyperlink does not work, also, keep it short and just call it the API page. Change the order such that the comment on the Wearipedia package comes first.

Also, does it even make sense to keep this paragraph anymore given that we now use the Wearipedia package?

**5. Adherence**

"that concisely describes when the participant has had their device on and off throughout the entirety of the time period" -> this is written for wearables and not a scale

**6. Simple viz**

What is pre-treatment and treatment?

**7. adv viz**

7.1

"First, it's important to know when the user has been wearing the device" -> this is written for wearable not a scale

"We'll interpret step count as a proxy for usage and plot colors accordingly."

You should comment on why we need this plot and then you should comment on what you see after plotting it.

7.2

A part of this plot is related to data cleaning. You should add it under that section.

**8. Data cleaning and outliers**

"We will use a simple statistical method for detecting outliers. Specifically, we will rely on the assumption that weight and fat ratio percentages roughly do not change very much from one timestep to the next, and that if there is one measurement that very clearly sticks up much higher than the rest surrounding it, then it is likely an outlier."

Is this what you measure? It seems to me that you consider all samples drawn from the same distribution, which doesn't take into account the time dimension. Instead, you could consider looking into the ratio between two weightings and require that they follow a normal distribution.

**9 Analysis**

In comparison to other notebooks, the synthetic data here has some labels on it. The previous notebooks discussed this, but it is not covered in this notebook. Please take some time out of the sections to discuss this dataset and what challenges it poses in an educational manner with reasonable plots.

9.1

A T-test assumes that data is sampled form a normal distribution, have you tested that assumption?

9.2

The first plot does not make sense to me, how is it convincing?
Also, please give some more intuition on how you are performing the testing.
And why are the two plots with different formatting?

cthev456 committed:
Merge branch 'Stanford-Health:main' into main

cthev456 committed:
Fixed comments + additional QOL changes

cthev456 commented:
Fixed the comments as well as addressed feedback from Thursday's meeting. Also added additional features such as an auto-start date setter for the visualization and minimized some code blocks.
